src/components/themed/TransactionListTop.tsx

```diff
@@ -113,99 +112,40 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
     super(props)
     this.state = {
       countryCode: undefined,
-      input: '',
-      lockedNativeAmount: '0',
-      stakePolicies: null,
-      stakePlugins: null,
-      stakePositionMap: {}
+      input: ''
     }
   }
 
   componentDidUpdate(prevProps: Props) {
     // Update staking policies if the wallet changes
     if (prevProps.wallet !== this.props.wallet) {
-      this.setState({ lockedNativeAmount: '0' })
-      this.updatePluginsAndPolicies().catch(err => showError(err))
+      this.props.dispatch(updateStakingState(this.props.currencyCode, this.props.wallet)).catch(err => showError(err))
     } else if (prevProps.tokenId !== this.props.tokenId) {
       // Update staked amount if the tokenId changes but the wallet remains the same
       let total = '0'
       let lockedNativeAmount = '0'
-      for (const stakePosition of Object.values(this.state.stakePositionMap)) {
+      for (const stakePosition of Object.values(this.props.walletStakingState.stakePositionMap)) {
         const { staked, earned } = getPositionAllocations(stakePosition)
         total = this.getTotalPosition(this.props.currencyCode, [...staked, ...earned])
         lockedNativeAmount = add(lockedNativeAmount, total)
       }
-      this.setState({ lockedNativeAmount })
+      this.props.dispatch({ type: 'STAKING/UPDATE_LOCKED_AMOUNT', walletId: this.props.wallet.id, lockedNativeAmount: lockedNativeAmount })
     }
   }
 
   componentDidMount() {
-    this.updatePluginsAndPolicies().catch(err => showError(err))
+    this.props.dispatch(updateStakingState(this.props.currencyCode, this.props.wallet)).catch(err => showError(err))
```

Comment:
Out of scope for this PR, but I wonder if we can be "smarter" about when we do these updates, now that the state management is more under control.